In [3]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv("../data/superstore_data.csv" , low_memory = False)

# Objective 
- To understand the dataset structure, analyze column datatypes, and prepare the data for further business analysis and visualization.



In [5]:
df.shape

(51290, 32)

In [6]:
df.duplicated().sum()

0

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 32 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   category           51290 non-null  object        
 1   city               51290 non-null  object        
 2   country            51290 non-null  object        
 3   customer_id        51290 non-null  object        
 4   customer_name      51290 non-null  object        
 5   discount           51290 non-null  float64       
 6   market             51290 non-null  object        
 7   column             51290 non-null  int64         
 8   order_date         51290 non-null  datetime64[ns]
 9   order_id           51290 non-null  object        
 10  order_priority     51290 non-null  object        
 11  product_id         51290 non-null  object        
 12  product_name       51290 non-null  object        
 13  profit             51290 non-null  float64       
 14  quanti

In [11]:
df.describe()

,discount,column,profit,quantity,sales,shipping_cost,order_year,order_month,order_quarter,order_day_of_week,is_weekend,order_week,sales_per_unit
count,51290.000000,51290.0,51290.000000,51290.000000,51290.000000,51290.000000,51290.000000,51290.000000,51290.000000,51290.000000,51290.000000,51290.000000,51290.000000
mean,0.142908,1.0,28.610982,3.476545,246.498440,26.375818,2012.777208,7.522636,2.793235,2.343264,0.108598,30.675687,71.659962
std,0.212280,0.0,174.340972,2.278766,487.567175,57.296810,1.098931,3.300056,1.066015,1.671486,0.311137,14.535168,113.581091
min,0.000000,1.0,-6599.978000,1.000000,0.000000,0.002000,2011.000000,1.000000,1.000000,0.000000,0.000000,1.000000,0.000000
25%,0.000000,1.0,0.000000,2.000000,31.000000,2.610000,2012.000000,5.000000,2.000000,1.000000,0.000000,20.000000,12.000000
50%,0.000000,1.0,9.240000,3.000000,85.000000,7.790000,2013.000000,8.000000,3.000000,2.000000,0.000000,33.000000,29.500000
75%,0.200000,1.0,36.810000,5.000000,251.000000,24.450000,2014.000000,10.000000,4.000000,4.000000,0.000000,44.000000,82.000000
max,0.850000,1.0,8399.976000,14.000000,22638.000000,933.570000,2014.000000,12.000000,4.000000,6.000000,1.000000,52.000000,3773.000000


# Initial Ovservation
- The dataset contains retail sales and shipping-related transactional information.
- a large dataset with 32 columns and 51290 records
- Date columns require datatype conversion before analysis.
- The dataset appears mostly clean with no immediate missing and duplicate values.


# Data cleaning

# Result:
- Date columns converted successfully to datetime datatype
- Different date formats required separate conversion logic
- Dataset structure is now more suitable for analy
- the column name column is waste so we drop itsis


In [8]:
df[['order_date','ship_date']].head()

,order_date,ship_date
0,07-01-2011,09-01-2011
1,21-01-2011,26-01-2011
2,05-08-2011,09-08-2011
3,05-08-2011,09-08-2011
4,29-09-2011,03-10-2011


In [10]:
df['order_date'] = pd.to_datetime(df['order_date'] , format ='%Y-%m-%d')
df['ship_date'] = pd.to_datetime(df['ship_date'] , format ='%Y-%m-%d')
df[['order_date', 'ship_date']].dtypes

order_date    datetime64[ns]
ship_date     datetime64[ns]
dtype: object

In [12]:
df['category'].unique()

array(['Office Supplies', 'Technology', 'Furniture'], dtype=object)

In [13]:
df.drop('column',axis = 1 ,inplace =True)

# Resolving profit margin 
- profit_margin contained invalid spreadsheet error values (#NAME?)
- Invalid entries were converted to NaN for proper numerical analysis

In [14]:
df.iloc[:,29].apply(type).value_counts()


profit_margin
<class 'str'>    51290
Name: count, dtype: int64

In [15]:
df['profit_margin'].unique()

array(['49.11', '48.91', '46.87', ..., '30.43', '34.19', '32.45'],
      dtype=object)

In [16]:
df['profit_margin'] = pd.to_numeric(df['profit_margin'], errors='coerce')


In [17]:
df['profit_margin'].dtype

dtype('float64')

In [18]:
df.to_csv('../data/cleaned_superstore_data.csv' , index =False)
